build-support/bin/check_pants_pex_abi.py

```diff
@@ -9,10 +9,7 @@
 import zipfile
 from pathlib import Path
 
-
-RED = "\033[31m"
-BLUE = "\033[34m"
-RESET = "\033[0m"
+from common import die, green
```

Comment:
This is definitely starting to get weird. Pants targets to allow linting, but a custom green function implemented as a copy of bash where we use ansicolors for said same in pants itself. And I think , etc... until the conversion to `./pants run ...` that you describe. I think the python3 porting is a bit premature and that's behind all this. Slightly bettr imo would be to port to `python_binary` targets first and then when pants can be python 3 only, do the fun stuff. All that said - you deserve a bit of joy after all the grind, so your call on ordering.

Thanks again for all your py3 work.

Reply:
> Slightly bettr imo would be to port to python_binary targets first and then when pants can be python 3 only, do the fun stuff.

Porting Pants itself to Py3 won't have any impact on whether we should implement our own color functions vs. use the colors library. We can today run the scripts either via `./pants run` or by directly invoking them, and Py3 compatibility works fine as is, even if we were to link it to our 3rd party deps and/or Pants src code.

The idea with solely depending on stdlib is it gives us the option to run directly, rather than requiring us to use `./pants run`. I think this is desirable for two reasons:

1) Closer alignment with how we run bash scripts. It shouldn't matter whether the build-support script is written in Bash or Python—that's an implementation detail. Requiring the user to use `./pants run` for one but not even allowing `./pants run` for the other (because Pants doesn't support bash targets) is surprising to me.

2) Ensures we have no dependencies on Pants, as some of the scripts are (will be) used to lint either the bash scripts or source code. For example, I'll soon be adding a [`shellcheck.py` script](https://github.com/pantsbuild/setup/blob/67ac92821b8031626b23b34a46494338b9de3152/build-support/shellcheck.py). If we only allow that script to be ran via `./pants run`, then let's say we have an issue with the `./pants` script that `shellcheck.py` is meant to check, the script won't be able to properly run shellcheck because the `./pants run` part will fail.
   * Although, now that I write this out, I'm less convinced this is important to maintain the separation. If `./pants run` fails because of an issue with the `./pants` script, presumably the error will print and the user will know they need to fix it? The only benefit I could see is if shellcheck would help to diagnose the issue. 

--

The duplication of `common.sh` is intentional. Imo, it should be just as easy to develop a Python script as a bash script, meaning we surface the same util code. There will be tasks that are better for Bash (e.g. installing Pyenv) and better for Python (e.g. `check_pants_pex_abi`, which started as Bash actually)—it should be well supported for the developer to choose the best language for the task at hand.

--

What are your thoughts on argument #1? I'm willing to use `colors` if you think #1 isn't as much of a concern as I fear. #2 isn't a huge deal it seems.

--

> All that said - you deserve a bit of joy after all the grind

Hehe thanks :)

Reply:
Ah one more reason I wanted to allow us to run Python scripts directly, rather than having to use `./pants run`: passthru command line flags get a lot less ergonomic when using `./pants run`.

Compare

```bash
$ build-support/bin/check_pants_pex_abi.py abi3 cp36m
``` 
to 

```bash
$ ./pants run build-support/bin:check_pants_pex_abi -- abi3 cp36m
```

Reply:
I guess I don't buy any of the direct-run arguments since we have some checks that only run via pants run no, like isort. These used to be scripted out of band. I really see nothing wrong with eating our dogfood here. If a non-human is calling run as is the case for your example, and extra matters little.

I think we just don't see eye to eye on these efforts and since you're wiling to put the work in - I'll desist in lodging minor tactical objections to how this work is being ordered.

Reply:
Sounds good - I updated the docstring to better address this conversation and will wait for feedback from others on this topic also. Thanks for bringing this all up.

One clarification.
>  I really see nothing wrong with eating our dogfood here

I completely agree dogfooding is valid, and often desirable, in these scripts. The sole exception I want to make is for `common.py`, because that is / will be used by the majority of the Python scripts, and I don't want to impose the 3 restrictions on all of those scripts in order to save some duplication of the colors library.

Individual scripts should certainly use 3rd party dependencies and Pants code, where appropriate. I updated the comment to explain this stdlib-only restriction only applies to `common.py`.

--

Also worth noting that I don't anticipate `common.py` growing any larger. I'm actively working to remove the complexity I added to the [setup repo's version](https://github.com/pantsbuild/setup/blob/gh-pages/build-support/common.py) by converting `ci.py` into a proper unit test file. I don't anticipate `common.py` ever having more functionality than what `common.sh` has. For example, if scripts need functionality from `dirutil.py`, those individual scripts will just directly depend on Pants.